packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -86,4 +134,131 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
             order_id: orderID,
         });
     }
+
+    private async _onHostedMethodApprove(data: ApproveDataOptions, actions: ApproveActions) {
+        try {
+            const orderDetails = await actions.order.get();
+            const consignments = this._store.getState().consignments.getConsignmentsOrThrow();
+            const lineItems = this._getLineItems();
+            if (!this._paymentMethod?.id) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+            const methodId = this._paymentMethod?.id;
+
+            if (this._currentShippingAddress) {
```

Comment:
Do we need to account for this? Since consignment should be updated on `_onShippingChangeHandler`?

Reply:
I added this condition because in order to update address or consignment we need `city, postal_code, country_code` from `this._currentshippingAddress` plus approved data about payer that we received  from paypal

Reply:
Yes I realised that, I was wonder more about do we need to update consignment as part of payment submit (which I assume is method approve)?

Reply:
if i get it right- we need to update it because we received real data about payer from paypal and need to pass it to update quote on back-end, in other case payment or even order will fail